### PR TITLE
[add-interceptor] Add request interceptor for hybrid development

### PR DESCRIPTION
**Pull request by aterribili**

### CODE REVIEW

File: Oahu/Oahu.swift [hunk 1]
```diff
@@ -45,4 +44,17 @@ public class Oahu: NSObject {
             request.setValue(header, forHTTPHeaderField: "Cookie")
         }
     }
+
+    public func webView(webView: WKWebView, decidePolicyForNavigationAction navigationAction: WKNavigationAction, decisionHandler: (WKNavigationActionPolicy) -> Void) {
+        print("url:\(navigationAction.request.URL?.absoluteString)")
```
Comment (fernandes456) [inline Oahu/Oahu.swift]: Remove this line

File: Oahu/OahuEvaluator.swift [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+public class OahuEvaluator: NSObject, Evaluator {
+    public var closure: () -> Void
+    public var url: String
+
+    public required init(closure: () -> Void, url: String) {
+        self.closure = closure
+        self.url = url
+    }
+}
```
Comment (fernandes456) [inline Oahu/OahuEvaluator.swift]: blank line

File: Oahu/Interceptor.swift [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    public func executeFirst(url: String) -> Bool {
+        if let first = self.evaluators.filter({$0.matchUrl(url)}).first {
+            first.execute()
+            return true
+        }
+
+        return false
+    }
+}
```
Comment (fernandes456) [inline Oahu/Interceptor.swift]: blank line

File: OahuTests/InterceptorTests.swift [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    func testShouldExecuteEvaluatorFromInterceptor() {
+        XCTAssertTrue(interceptor.executeFirst("http://www.bing.com"))
+    }
+
+    func testShouldExecuteFirstEvaluatorFromInterceptor() {
+        XCTAssertTrue(interceptor.executeFirst("https://www.google.com"))
+    }
+}
```
Comment (fernandes456) [inline OahuTests/InterceptorTests.swift]: blank line

File: OahuTests/EvaluatorTests.swift [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+        XCTAssertFalse(evaluator.matchUrl("http://www.yahoo.com"))
+    }
+
+    func testShouldValidateEvaluatorClosureExecution() {
+        let evaluator = OahuEvaluator(closure: {XCTAssertTrue(true)}, url: "www.google.com")
+        
+        evaluator.execute()
+    }
+}
```
Comment (fernandes456) [inline OahuTests/EvaluatorTests.swift]: blank line

Comment (aterribili) [inline OahuTests/EvaluatorTests.swift]: done :+1:

File: Oahu/OahuEvaluator.swift (hunk 2 above)
Comment (aterribili) [inline Oahu/OahuEvaluator.swift]: done :+1:

File: Oahu/Interceptor.swift (hunk 3 above)
Comment (aterribili) [inline Oahu/Interceptor.swift]: It's means add blank endline :smile: 

done :+1:

File: OahuTests/InterceptorTests.swift (hunk 4 above)
Comment (aterribili) [inline OahuTests/InterceptorTests.swift]: :+1: done

**Pull request merged**